src/Input/Textarea/Textarea.test.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import Textarea from './Textarea';
+import { SecondaryColor, PrimaryColor } from '../../Style/Colors';
+
+const props = {
+  label: 'Description',
+  onChange: jest.fn().mockImplementation(event => event.target.value),
+};
+
+const inputValue =
+  'As the first sign in the zodiac, the presence of Aries always marks the beginning of something energetic and turbulent.';
+
+function setupTextarea(otherProps?: any) {
+  const { getByLabelText, ...utils } = render(
+    <Textarea label={props.label} onChange={props.onChange} {...otherProps} />
+  );
+  const textareaInput = getByLabelText(props.label) as HTMLTextAreaElement;
+  return { textareaInput, utils };
+}
+
+function setupTextareaWithChangeEvent(otherProps?: any) {
+  const { textareaInput } = setupTextarea(otherProps);
+  userEvent.type(textareaInput, inputValue);
+  return { textareaInput };
+}
+
+function createNodeMock(element: any) {
+  if (element.type === 'textarea') {
+    return {
+      value: 0,
+      offsetHeight: 34,
+    };
+  }
+  return null;
+}
+
+it('<Textarea> should render with a label of Description', () => {
+  const options = { createNodeMock };
+  const TextareaSnapshot = renderer
+    .create(<Textarea label={props.label} />, options)
+    .toJSON();
+  expect(TextareaSnapshot).toMatchSnapshot();
+});
+
+describe('when it is rendered', () => {
+  it('should display the correct input label', () => {
+    const { textareaInput } = setupTextarea();
+    expect(textareaInput).toBeTruthy();
```

Comment:
Hi @franciscrispin do you follow any practice for this kind of test?

look at this test case, I don't know which input is correct in this textarea because the value is in setupTextarea function
and this value should never return false value right 🤔 ?

could we have something like:
1. define arguments
2. get the component based on the arguments
3. check the component fields

Reply:
Oh in this case `const textareaInput = getByLabelText(props.label) as HTMLTextAreaElement;` will error if the `value` is not found so `expect(textareaInput).toBeTruthy();` will be `falsy` and the test will fail. 

I don't really have a testing practice like you described haha. Can you share an example of the 3 steps you described? @aaa59891

Reply:
Okay thank you! Will look into it!